src/sentry/static/sentry/app/components/group/externalIssueActions.tsx

```diff
@@ -37,87 +37,146 @@ class ExternalIssueActions extends AsyncComponent<Props, State> {
     this.state = {
       showModal: false,
       action: 'create',
-      selectedIntegration: this.props.integration,
-      issue: this.getIssue(),
+      selectedIntegration: null,
       ...this.getDefaultState(),
+      ...this.linkedIssuesFilter(),
     };
   }
 
   getEndpoints() {
     return [];
   }
 
-  getIssue() {
-    return this.props.integration && this.props.integration.externalIssues
-      ? this.props.integration.externalIssues[0]
-      : null;
+  linkedIssuesFilter() {
+    return this.props.configurations.reduce(
+      (acc, curr) => {
+        if (curr.externalIssues.length) {
+          acc.linked.push(curr);
```

Comment:
@NisanthanNanthakumar Do we even need to add `linked` and `unlinked` for state? Couldn't you just determine them with getters from your `configurations`? Better to have minimal state. The number of configurations a user have should be small so I wouldn't be concerned about calculating them each re-render.

Reply:
@scefali we have this state bc when a user creates/deletes a linked issue, we don't refetch the new configurations from the server. Instead we have to mutate the current state with the returned values. If a user creates and deletes multiple times, `this.props.configurations` will be out of date. It will still have the values from when we first loaded the page.

Reply:
@NisanthanNanthakumar How hard would it be to update the props with a callback? I'm a bit concerned if we have prop values that are out of date as it might cause a problem with other components used in `ExternalIssueList`.  If we can't do that, I think it would be easier to just have one state variable in `ExternalIssueActions` for the configurations and update that when we need to make state changes.

Reply:
@scefali we don't use `configurations` elsewhere in `ExternalIssueList`. It is only calculated for our integrations. Furthermore, `ExternalIssueActions` is a component that's only used in sentry to render the integrations in `ExternalIssueList` component. 

So there is no need to bring the state up to the parent bc we don't need it for any of the sibling components.

Reply:
@NisanthanNanthakumar sure, but we should future proof our components as much as possible and use minimal state when it makes sense. What we have now is excessively complicated IMO.